### DISCUSSION TITLE
Flash loan integration

### DISCUSSION THREAD

**Discussion by Fluffy9** [PaulRBerg/prb-proxy#24] [Ideas]
I'm really loving this project! 

I think flash loan or flash swap support would really make this project superior to DSProxy. That wasn't even available when the original DSProxy code came out. I'm still learning solidity, but I'd like to contribute

**Comment by PaulRBerg** [comment]
Hey @Fluffy9, glad you like it!

I don't think I understand what you mean by "flash swap support". This proxy is meant to delegate calls to target contracts, which can contain arbitrary logic, which can be about performing flash swaps.

Furthermore, each flash swap provider has different integration logic. I wouldn't like this proxy to be specific to one project, say, specific to Uniswap v3.

**Reply by Fluffy9** [reply]
Sorry, something like this. https://eips.ethereum.org/EIPS/eip-3156

**Reply by PaulRBerg** [reply]
At a glance it seems to me that if a target contract implements EIP-3156 then the proxy can use it that way. I don't see why I should implement it in the proxy contract. Do you know how [DELEGATECALL](https://ethereum.stackexchange.com/questions/3667/difference-between-call-callcode-and-delegatecall) works?

**Reply by Fluffy9** [reply]
I understand that prb-proxy could be used to delegatecall another contract and that contract could be executing something that implements flashloans. That's how DefiSaver works and probably InstaDapp too. I think it would be really useful if that could be implemented on the proxy contract level instead since it's commonly needed and the EIP has a common format for it.

**Reply by PaulRBerg** [reply]
The proxy is meant to do one thing only, which is to delegate call to target contracts. Stuffing the flash loan logic in the core contract would add up to the deployment gas cost, which is already quite high.

**Reply by cleanunicorn** [reply]
@Fluffy9 I think plugin support https://github.com/paulrberg/prb-proxy/discussions/42 would help with this without increasing the gas cost (when it's not enabled).

**Comment by PaulRBerg** [comment]
#42 has been merged in, so PRBProxy now offers support for flash loan callbacks via the plugin system.
